robot/impl/robot_reconfigure_test.go

```diff
@@ -3540,9 +3541,10 @@ func TestResourceConstructCtxCancel(t *testing.T) {
 				API:   mockAPI,
 			},
 			{
-				Name:  "two",
-				Model: model1,
-				API:   mockAPI,
+				Name:      "two",
+				Model:     model1,
+				API:       mockAPI,
+				DependsOn: []string{"one"},
```

Comment:
this tests configure cancellations - it's intended to flow as follows:

1. configure new robot
2. cancel context immediately
3. assert that only one of the two resources got constructed

now that we construct resources in parallel when possible, it's more likely that both resources will get configured before the cancellation is detected. to get around this, we make one resource depend on the other simulate synchronous configuration.

let me know if this warrants a code comment, or if there is a more clear way to indicate the intention of this test

Reply:
a comment would be good